extensions/stackdriver/stackdriver.cc

```diff
@@ -54,8 +56,23 @@ void StackdriverRootContext::onConfigure(
     return;
   }
 
-  // Only register exporter once in main thread when initiating base WASM
-  // module.
+  // Get node metadata. GetMetadataStruct always returns the whole node metadata
+  // even with a key passed in.
+  // TODO: clean this up after fixing upstream API to respect node metadata key.
+  auto node_metadata = getMetadataStruct(Common::Wasm::MetadataType::Node, "");
+  auto istio_metadata = node_metadata.fields().find(kIstioMetadataKey);
```

Comment:
you can use
```c++
getMetadataValue(Common::Wasm::MetadataType::Node, "istio.io/metadata");
```

Reply:
I don't think it respects the key per the comment and my own experience.

Reply:
Can you check what you are using in metadata exchange ?
Is it trying to get the whole metadata or just our sub key?

Reply:
I'm using low-lever proxy_get/setMetadataStruct. The problem with C++ version is that it automatically serializes the whole struct, and I didn't want to since I have bytes already.

Reply:
I think we need a vararg version of this method that can traverse into structs. Basically, something along Find() in https://github.com/envoyproxy/envoy-wasm/issues/42. That way there is no transfer of container objects to get a value.

Reply:
Done.